dashboard/app/controllers/learning_goal_evaluations_controller.rb

```diff
@@ -25,6 +24,31 @@ def update
     end
   end
 
+  def get_evaluation
+    learning_goal_evaluation = LearningGoalEvaluation.find_by(
+      user_id: learning_goal_evaluation_params[:user_id],
+      teacher_id: current_user.id,
+      learning_goal_id: learning_goal_evaluation_params[:learning_goal_id]
+    )
+
+    if learning_goal_evaluation&.teacher_id == current_user.id
+      render json: learning_goal_evaluation
+    else
+      head :not_found
+    end
+  end
+
+  def get_or_create_evaluation
+    learning_goal_evaluation = LearningGoalEvaluation.find_or_create_by(
+      user_id: learning_goal_evaluation_params[:user_id],
+      teacher_id: current_user.id,
+      learning_goal_id: learning_goal_evaluation_params[:learning_goal_id]
+    )
+
+    learning_goal_evaluation.save!
```

Comment:
is this needed?

Reply:
Not sure. This could be rails ignorance on my part, but I assumed that if you create a new evaluation, it needs to be saved.

Reply:
`create` actually inserts the row into the database so I believe the `save` here is redundant ([create documentation](https://apidock.com/rails/v4.0.2/ActiveRecord/Relation/create) which is used by [find_or_create](https://apidock.com/rails/v4.0.2/ActiveRecord/Relation/find_or_create_by))